chatbot-core/api/services/memory.py

```diff
@@ -0,0 +1,57 @@
+"""
+Handles in-memory chat session state (conversation memory).
+Provides utility functions for session lifecycle.
+"""
+
+import uuid
+from langchain.memory import ConversationBufferMemory
+
+# sessionId --> history
+_sessions = {}
+
+def init_session() -> str:
```

Comment:
ig we are going to use this in a FastAPI or multithreaded environment, so, _sessions dictionary can become a race condition. what i suggest is import Lock 
```
_sessions = {}
_lock = Lock()

def init_session() -> str:
    session_id = str(uuid.uuid4())
    with _lock:
        _sessions[session_id] = ConversationBufferMemory(return_messages=True)
    return session_id ``` 

Similarly wrap get, delete, and exists